test/upgrade/autoscaler.go

```diff
@@ -41,14 +42,19 @@ func AutoscaleSustainingTest() pkgupgrade.BackgroundOperation {
 	var ctx *e2e.TestContext
 	var wait func() error
 	stopCh := make(chan time.Time)
+	var testStreamCancel logstream.Canceler = func() {}
+	var sysStreamCancel logstream.Canceler = func() {}
 	return pkgupgrade.NewBackgroundVerification("AutoscaleSustainingTest",
 		func(c pkgupgrade.Context) {
 			// Setup
-			ctx = e2e.SetupSvc(c.T, autoscaling.KPA, autoscaling.Concurrency, containerConcurrency, targetUtilization,
+			ctx = e2e.SetupSvc(c.T, "autoscale-sustaining", autoscaling.KPA, autoscaling.Concurrency, containerConcurrency, targetUtilization,
 				rtesting.WithConfigAnnotations(map[string]string{
 					autoscaling.TargetBurstCapacityKey: "0", // Not let Activator in the path.
 				}))
-			ctx.SetLogger(c.Log.Infof)
+			if !test.ServingFlags.DisableLogStream {
```

Comment:
Since we are repeating this would it make sense to have something like `NewBackgroundOperationWithLogStream`?

Reply:
That would probably complicate things as the logstream setup needs to know the ksvc name which is only created inside the handler.

Reply:
Yeah. But the logstream from e2e.Setup prints nothing as it filters the lines according to a wrong service name which doesn't appear in the logs actually. I though this would be alright.
Just noticed that for the Prober test we can use something like this to disable the logstream from e2e.Setup:
```
clients = test.Setup(c.T, test.Options{DisableLogStream: true})
```
But for the Autoscale tests it's more complicated. I'd have to add another parameter to e2e.SetupSvc (called e.g. disableLogStream) and then pass it to test.Setup(c.T, test.Options{DisableLogStream: disableLogStream). 
Another option would be to store the original value, call e2e.SetupSvc with disabled log stream and then use the original value again (and it's ugly):
```
                        disableLogStreamFlag := test.ServingFlags.DisableLogStream
			test.ServingFlags.DisableLogStream = true
			ctx = e2e.SetupSvc(c.T, autoscaling.KPA, autoscaling.Concurrency, containerConcurrency, targetUtilization,
				rtesting.WithServiceName("autoscale-sus-tbc"),
				rtesting.WithConfigAnnotations(map[string]string{
					autoscaling.TargetBurstCapacityKey: "-1", // Put Activator always in the path.
				}))
			if !disableLogStreamFlag {
				canceler = streamLogs(c.T, ctx.Clients(), ctx.Names().Service)
			}
```